hugegraph-server/hugegraph-dist/src/assembly/static/bin/preload-topling.sh

```
@@ -0,0 +1,108 @@
+    }
+    require_topling_platform || exit 1
+
+    TOPLING_JAR=$(ls -1 "$SERVER_LIB"/topling/rocksdbjni*.jar 2>/dev/null | sort -V | tail -1 || true)
+    if [ -z "${TOPLING_JAR:-}" ]; then
+        echo "Error: rocksdb.provider=topling but no ToplingDB JAR found in $SERVER_LIB/topling/" >&2
+        echo "       Please place the ToplingDB rocksdbjni JAR in lib/topling/ directory." >&2
+        exit 1
+    fi
+
+    # Swap: move standard JAR aside, copy ToplingDB JAR into lib/
+    STANDARD_JAR=$(ls -1 "$SERVER_LIB"/rocksdbjni*.jar 2>/dev/null | sort -V | tail -1 || true)
```

Shellcheck (0.11.0)

[info] Use find instead of ls to better handle non-alphanumeric filenames.

(SC2012)